### DISCUSSION TITLE
Feature request: Define number of speaks for diarization

### DISCUSSION THREAD

**Discussion by pdjewell** [deepgram/community#217] [Feature Requests]
Firstly I'd like to say how much I am loving deepgram and how easy it is to use!

I was wondering if it would be possible to add a feature. Would it be possible to define the number of speakers there are, which could therefore improve the accuracy of diarization? I know this can be done with pyannote for example. Thanks!

**Comment by incubatize** [comment]
@pdjewell  - thank you for your feedback. It helps us to improve user experience with our product.
Just to clarify - you'd like to specify exact number of speakers or max number of speakers that a call can have?

**Reply by erikmunson** [reply]
Big, huge +1 to this. Assembly AI has this on their transcription API: https://www.assemblyai.com/docs/models/speaker-diarization#specifying-the-number-of-speakers

This is EXTREMELY high value if you are doing something like transcribing a video meeting where you know for a fact how many people were on the call, and want to constrain the output of the diarization to match the number of speakers you know are in the call. The fact that this feature is in the Assembly API and not deepgram is one of the primary reasons we use Assembly at my company, we would seriously consider switching if Deepgram had it. When evaluating Deepgram we routinely saw more speakers identified in the diarization than were actually on the calls in our test dataset.

**Comment by lukeocodes** [comment]
duplicate of https://github.com/orgs/deepgram/discussions/176
